components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsBindingTest.java

```diff
@@ -18,32 +18,69 @@
+import java.util.Date;
 
 import org.apache.activemq.command.ActiveMQBlobMessage;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.when;
 
+@RunWith(MockitoJUnitRunner.class)
 public class JmsBindingTest {
 
+    private final Instant instant = Instant.ofEpochMilli(1519672338000L);
+
+    @Mock
+    private JmsConfiguration mockJmsConfiguration;
+    @Mock
+    private JmsEndpoint mockJmsEndpoint;
+
+    private JmsBinding jmsBindingUnderTest;
+
+    @Before
+    public void setup() {
+        when(mockJmsConfiguration.isFormatDatePropertiesToIso8601()).thenReturn(false);
+        when(mockJmsConfiguration.isMapJmsMessage()).thenReturn(true);
+        when(mockJmsEndpoint.getConfiguration()).thenReturn(mockJmsConfiguration);
+        jmsBindingUnderTest = new JmsBinding(mockJmsEndpoint);
+    }
+
     @Test
-    public void testJmsBindingNoArgs() throws Exception {
-        JmsBinding underTest = new JmsBinding();
-        assertNull(underTest.extractBodyFromJms(null, new ActiveMQBlobMessage()));
+    public void testExtractNullBodyFromJmsShouldReturnNull() throws Exception {
+        assertNull(jmsBindingUnderTest.extractBodyFromJms(null, new ActiveMQBlobMessage()));
     }
 
     @Test
-    public void testGetValidJmsHeaderValueWithBigInteger() {
-        JmsBinding binding = new JmsBinding();
-        Object value = binding.getValidJMSHeaderValue("foo", new BigInteger("12345"));
+    public void testGetValidJmsHeaderValueWithBigIntegerShouldSucceed() {
+        Object value = jmsBindingUnderTest.getValidJMSHeaderValue("foo", new BigInteger("12345"));
         assertEquals("12345", value);
     }
 
     @Test
-    public void testGetValidJmsHeaderValueWithBigDecimal() {
-        JmsBinding binding = new JmsBinding();
-        Object value = binding.getValidJMSHeaderValue("foo", new BigDecimal("123.45"));
+    public void testGetValidJmsHeaderValueWithBigDecimalShouldSucceed() {
+        Object value = jmsBindingUnderTest.getValidJMSHeaderValue("foo", new BigDecimal("123.45"));
         assertEquals("123.45", value);
     }
+
+    @Test
+    public void testGetValidJmsHeaderValueWithDateShouldSucceed() {
+        Object value = jmsBindingUnderTest.getValidJMSHeaderValue("foo", Date.from(instant));
+        assertNotNull(value);
+        // We can't assert further as the returned value is bound to the machine time zone and locale
+    }
+
+    @Test
+    public void testGetValidJmsHeaderValueWithIso8601DateShouldSucceed() {
+        when(mockJmsConfiguration.isFormatDatePropertiesToIso8601()).thenReturn(true);
+        Object value = jmsBindingUnderTest.getValidJMSHeaderValue("foo", Date.from(instant));
+        assertEquals("2018-02-26T19:12:18Z", value);
```

Comment:
Expressing date at GMT+0 is not necessarily easy to read for end users deploying in a single country. But, I find them easier to work with when your deployments is spread across data centers located in different time zones.